src/Bicep.LangServer/Snippets/ResourceSnippetsProvider.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Bicep.LanguageServer.Snippets
+{
+    public class ResourceSnippetsProvider : IResourceSnippetsProvider
+    {
+        private static readonly Dictionary<string, string> _snippetNameToDetailMap = new Dictionary<string, string>()
+        {
+            { "res-aks-cluster", "Kubernetes Service Cluster" },
+            { "res-app-security-group", "Application Security Group" },
+            { "res-automation-account", "Automation Account" },
+            { "res-availability-set", "Availability Set" },
+            { "res-container-group", "Container Group" },
```

Comment:
>res [](start = 15, length = 3)

Probably out of scope for this PR, but I wonder if we could something to place the detail strings in the snippet files. It could take the form of a special structured comment at the top of the file, a custom decorator or something like that.

Reply:
@majastrz, I moved description to bicep(added comment basically). Could you please take a look?
https://github.com/Azure/bicep/commit/144941b7b682fca24accb32deae40dd20530307b

@StephenWeatherford , I made sure there are no errors in the bicep files that are part of this pr. I'll check if I can add a test

Reply:
I'll be adding test to validate snippet contents in a follow up PR.